chain/stmgr/searchwait.go

```diff
@@ -252,22 +252,27 @@ func (sm *StateManager) tipsetExecutedMessage(ts *types.TipSet, msg cid.Cid, vmm
 
 		if m.VMMessage().From == vmm.From { // cheaper to just check origin first
 			if m.VMMessage().Nonce == vmm.Nonce {
-				if allowReplaced && m.VMMessage().EqualCall(vmm) {
-					if m.Cid() != msg {
-						log.Warnw("found message with equal nonce and call params but different CID",
-							"wanted", msg, "found", m.Cid(), "nonce", vmm.Nonce, "from", vmm.From)
-					}
+				if !m.VMMessage().EqualCall(vmm) {
+					// this is an entirely different message, fail
+					return nil, cid.Undef, xerrors.Errorf("found message with equal nonce as the one we are looking for that is NOT a valid replacement message (F:%s n %d, TS: %s n%d)",
+						msg, vmm.Nonce, m.Cid(), m.VMMessage().Nonce)
+				}
 
-					pr, err := sm.cs.GetParentReceipt(ts.Blocks()[0], i)
-					if err != nil {
-						return nil, cid.Undef, err
+				if m.Cid() != msg {
+					log.Warnw("found message with equal nonce and call params but different CID",
```

Comment:
should only warn if we _don't_ allow replaced. Otherwise, the user will see scary messages.

Reply:
fair